heaps/heap_sort.py

```diff
@@ -1,8 +1,41 @@
+from heapq import heappush, heappop
+from heaps.min_heap import MinHeap
 
-
-def heap_sort(list):
+def heap_sort(list): # O(n)
     """ This method uses a heap to sort an array.
-        Time Complexity:  ?
-        Space Complexity: ?
+        Time Complexity:  O(n log n)
+        Space Complexity: O(n)
     """
-    pass
\ No newline at end of file
+    heap = MinHeap()
+    for num in list:
+        # has heap_up inside and sorts it in heap everytime new node is added
+        heap.add(num)
+
+    # while the heap is not empty, it removes the most minimal value which is always the root
+    index = 0
+    while not heap.empty():
+        # puts sorted nums back in list using index
+        # remove is O(log n)
+        list[index] = heap.remove()
+        index += 1
+
+    return list
```

Comment:
Note the since this isn't a fully in-place solution (the MinHeap has a O(n) internal store), we don't necessarily need to modify the passed in list. The tests are written to check the return value, so we could unpack the heap into a new result list to avoid mutating the input.

Also, in this situation, since we built the heap, we also "know" the number of items in the heap. So it's OK to iterate a fixed number of times. But if we were pulling things out of a heap more generally, we would want to make use of the `empty` helper as follows:

```py
    result = []
    while not heap.empty():
        result.append(heap.remove())

    return result
```